lib/lz4.c

```diff
@@ -1038,6 +1066,8 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
             } else
                 *token += (BYTE)(matchCode);
         }
+        /* Ensure we have enough space for the last literals. */
+        assert(!(outputDirective == fillOutput && op + 1 + LASTLITERALS > olimit));
```

Comment:
This assert isn't quite correct (or is exposing a bug) so I'll look into it.

Reply:
I believe it is triggered because the condition on line 1030 isn't quite right. As I understand it `matchCode>>8` has an off by one error in the division. Also, we could need one extra zero byte.

I think we need something like `(matchCode+240)/255`.

Reply:
This assert was failing when `matchCode == 7665`. `matchCode >> 8 == 29`, but it actually needs 31 bytes since `(matchCode + 240) / 255 == 31` (the extra byte is needed because 7665 - 15 is a multiple of 255).